### ISSUE TITLE
Issue: CMP is being blocked by uBlock Origin

### ISSUE THREAD

**Issue by nmpina28** [uBlockOrigin/uAssets#8588]
### Description

Greetings,
My name is Nelson Pina and I work for an Ad Tech company named Snigel (snigel.com) that provides services related to website ad monetization, ad optimization, revenue management and optimization (machine learning), and GDPR compliant CMP implementations (TCF 2.0 compliant), according to the IAB requirements: `https://iabeurope.eu/transparency-consent-framework/ `
I am contacting the support of **uBlock Origin** because I need your help in solving an issue related to your product and the display of our CMP, causing a policy violation regarding GDPR's requirements.

First, I have to state that our CMP is registered at IAB (`https://iabeurope.eu/`) which means that meets 100% of IAB's mandatory requirements. Second, I can say that the implementation of our CMP by our publishers is 100% in accordance with the guidelines of the IAB. This being said, I will now explain the problem caused by [Ad Block name]. If a user accesses a site with our CMP implemented and no previous consent has been given or collected, the following Modal Window, or a Banner, with all the necessary legal information should be displayed asking for consent:

![image](https://user-images.githubusercontent.com/31660478/108249438-fc57d580-714c-11eb-94da-90022d2caf11.png)

But instead, no CMP appears. Your product, **uBlock Origin** is blocking our IAB registered CMP and preventing any consent collection, incurring policy violations regarding GDPR, necessary not only for displaying Ads but for e.g. custom publisher consent. In the following image, it is evident that no advertisement is shown because no consent was given, which should have been provided by our CMP.

![image](https://user-images.githubusercontent.com/31660478/108249493-0bd71e80-714d-11eb-8425-e05f620cc686.png)

After consulting the IAB, we were told to get in contact with you and make you aware of the problems, and to ask you to whitelist the following domain: ***.snigelweb.com/***
Not sure how this domain has been ending up on your blocklist, but this is a clear false positive and I´d like to ask you to remove it and whitelist it again.

Please confirm if you agree and by when the domain will be whitelisted.

Any questions, please don´t hesitate to reach out to me.

Nelson Pina
--

### A specific URL where the issue occurs

`https://www.morewords.com/`

### Steps to Reproduce

1. Activate the uBlock origin extension on the browser
2. Access a page that uses our CMP.
3. The CMP isn't displayed

### Expected behavior:

The CMP window or banner should be displayed so the user can customize his consent and privacy choices and accept. 

### Actual behavior:

The CMP isn't displayed.

### Your environment

* uBlock Origin version: This issue is not caused by a specific version, and happens even if this is the only extension installed.
* Browser Name and version: All supported browsers by uBlock Origin
* Operating System and version: OS independent

**Comment by gorhill** [comment]
> I work for an Ad Tech company named Snigel (snigel.com)

Filter issues are not to be reported on uBO's repo, this is a filter issue, the site tries to load a 3rdparty script from `snigelweb.com`, which I assume is related to your company. The request is blocked by EasyList:

    ||snigelweb.com^$3p

I trust that this the filter works as intended, so closing as wontfix. If you want to dispute the filter, you will have to ask EasyList maintainers: <https://github.com/easylist/easylist/issues>.

**Comment by llacb47** [comment]
FYI @nmpina28 

Across four different sites:
```
https://www.morewords.com/
https://www.wordhippo.com/
https://www.w3schools.com/
https://www.linguee.com/
```

I had to add this many rules to unbreak the consent dialog:
```
@@||snigelweb.com/adengine/*/loader.js$script
@@||snigelweb.com/*/adngin.js$script
@@||snigelweb.com/adconsent/
@@||snigelweb.com/gvl/
@@||snigelweb.com/sncmp/
@@||snigelweb.com/pub/*/snhb$script
@@||snigelweb.com/snhb/
@@||geo-*.snigelweb.com/bp/
```

the `hb` in `snhb` obviously stands for header bidding... IMO it is unacceptable to be asked to allow ad scripts for the purposes of GDPR consent. I'm 99% easylist will agree.

**Comment by uBlock-user** [comment]
Also without creating any exceptions, don't see breakage either.

**Comment by llacb47** [comment]
> Also without creating any exceptions, don't see breakage either.

You have to clear cookies on one of those sites and visit from an EU ip. You should see a consent popup but it's blocked by EL

**Comment by nmpina28** [comment]
Thank you for your fast reply. But this should issue should be handled in a more private way. Besides, it's of my knowledge that the CMP should not be blocked in any case.

**Comment by nmpina28** [comment]
> 
> Filter issues are not to be reported on uBO's repo, this is a filter issue, the site tries to load a 3rd-party script from `staging-cdn.snigelweb.com/adengine/`, which I assume is related to your company. The request is blocked by EasyList:
> 
> ```
> ||snigelweb.com^$3p
> ```
> 
> I trust that this the filter works as intended (i.e. not a false positive), so closing as wontfix. If you want to dispute the filter, you will have to ask EasyList maintainers: https://github.com/easylist/easylist/issues.

Thank you for your help on this subject. And for your suggestions. I'll test it again. Can I ask you to delete all your previous comments. Thank you and all you explained was very useful
